toolchain/check/handle_variable.cpp

```diff
@@ -26,24 +26,18 @@ auto HandleReturnedModifier(Context& context, Parse::NodeId parse_node)
 
 auto HandleVariableInitializer(Context& context, Parse::NodeId parse_node)
     -> bool {
-  // No action, just a bracketing node.
-  context.node_stack().Push(parse_node);
+  SemIR::InstId init_id = context.node_stack().PopExpr();
+  context.node_stack().Push(parse_node, init_id);
   return true;
 }
 
 auto HandleVariableDecl(Context& context, Parse::NodeId parse_node) -> bool {
   // Handle the optional initializer.
   auto init_id = SemIR::InstId::Invalid;
   Parse::NodeKind next_kind = context.node_stack().PeekParseNodeKind();
-  if (next_kind == Parse::NodeKind::TuplePattern) {
-    return context.TODO(parse_node, "tuple pattern in var");
-  }
-  // TODO: find a more robust way to determine if there was an initializer.
-  bool has_init = next_kind != Parse::NodeKind::BindingPattern;
+  bool has_init = next_kind == Parse::NodeKind::VariableInitializer;
   if (has_init) {
     init_id = context.node_stack().PopExpr();
-    context.node_stack()
-        .PopAndDiscardSoloParseNode<Parse::NodeKind::VariableInitializer>();
   }
```

Comment:
```suggestion
  auto init_id = context.node_stack().PopIf<Parse::NodeKind::VariableInitializer>();
```

Reply:
I'd assumed this code was intentionally avoiding `std::optional` for performance reasons, but that works for me!

Reply:
We could alternatively use `.value_or(SemIR::InstId::Invalid)` here but I doubt it'll make any real difference.